core/station/impl/src/models/external_canister.rs

```diff
@@ -219,6 +231,176 @@ impl ModelValidator<ExternalCanisterError> for ExternalCanister {
     }
 }
 
+fn assert_policy_is_associated_with_target_canister<F>(
+    policy_id: &UUID,
+    is_associated: F,
+) -> ModelValidatorResult<ExternalCanisterValidationError>
+where
+    F: Fn(&RequestPolicy) -> bool,
+{
+    let policy = REQUEST_POLICY_REPOSITORY.get(policy_id).ok_or_else(|| {
+        ExternalCanisterValidationError::ValidationError {
+            info: format!(
+                "The request policy with id '{}' does not exist.",
+                Uuid::from_bytes(*policy_id).hyphenated()
+            ),
+        }
+    })?;
+
+    // validates if the policy matches the expected variant and canister_id or throws an error
+    if !is_associated(&policy) {
+        Err(ExternalCanisterValidationError::ValidationError {
+            info: format!(
+                "The request policy with id '{}' is not associated with the external canister.",
+                Uuid::from_bytes(*policy_id).hyphenated()
+            ),
+        })?;
+    }
+
+    Ok(())
+}
+
+fn validate_change_policies_are_associated_with_target_canister(
+    canister_id: &Principal,
+    policy_ids: &[UUID],
+) -> ModelValidatorResult<ExternalCanisterValidationError> {
+    for policy_id in policy_ids {
+        // validates if the policy matches the expected change variant and canister_id or throws an error
+        assert_policy_is_associated_with_target_canister(policy_id, |policy| {
+            matches!(
+                policy.specifier,
+                RequestSpecifier::ChangeExternalCanister(ExternalCanisterId::Canister(id)) if id == *canister_id
```

Comment:
How would an admin set a request policy for all canisters ExternalCanisterId::Any?

Reply:
Using `RequestOperation::EditRequestPolicy`?